src/autoscaler/eventgenerator/generator/evaluator_test.go

```diff
@@ -579,11 +579,12 @@ var _ = Describe("Evaluator", func() {
 					database.RetrieveAppMetricsStub = func(appId string, metricType string, start int64, end int64) ([]*models.AppMetric, error) {
 						return appMetricGTBreach, nil
 					}
-					Expect(triggerChan).To(BeSent(triggerArrayGT))
 				})
 				Context("when the send request encounters error", func() {
 					JustBeforeEach(func() {
 						scalingEngine.Close()
+						Eventually(scalingEngine.HTTPTestServer).Should(BeNil())
```

Comment:
This isn't thread safe. Since the goal is to guarantee the scaling engine is not responding, we should be sending a request until it fails.